assets/javascript/apps/pipeline/nodes/GetInputWidget.tsx

```diff
@@ -0,0 +1,196 @@
+import {
+  HistoryNameWidget,
+  HistoryTypeWidget,
+  KeywordsWidget,
+  LlmModelWidget,
+  LlmProviderIdWidget,
+  MaxTokenLimitWidget,
+  SourceMaterialIdWidget,
+  TextWidget
+} from "../widgets";
+import React from "react";
+import {getCachedData} from "../utils";
+import {InputParam} from "../types/nodeInputTypes";
+import {NodeParams} from "../types/nodeParams";
+
+interface InputWidgetParams {
```

Comment:
Is there a particular reason you chose to use interface over a type for this?

Reply:
I did a bit of digging on when to use interface vs type aliases. [This article](https://www.totaltypescript.com/type-vs-interface-which-should-you-use) was insightful, but reaches a different conclusion to the typescript team ([typescript docs](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces)). Interesting. Looks like we should be fine using either, but it's good to know the caveats mentioned in the first article

Reply:
Thanks, that's useful. `type` definitely seems preferable in this instance.

Reply:
81c2b2ffa6c18648f49631d99d2166c1c3f8c864